### DISCUSSION TITLE
Question about using spacy pipline for NER, coref and relation extraction models

### DISCUSSION THREAD

**Discussion by wendywtchang** [explosion/spaCy#11290] [Help: Model Advice]
Hello,

We want to use spacy models to build a NER, Coref, and relation extraction pipeline. 
Could you suggest us a way to stack them together using the spacy pipeline? Especially the ordering of these three models.

**Comment by polm** [comment]
If you're using our example relation extraction component, that would typically rely on the NER labels, so it would have to come after that. It depends on how you generate candidates for relation extraction though.

What coref component are you using? The one we're working on (that isn't quite released yet) doesn't use labels of other components, so where it falls in the order wouldn't matter, unless you use it to generate candidates for relation extraction. 

Other than that, there's the question of whether you train your components all at once or separately and then assemble them. The general overview of that is covered [here](https://spacy.io/usage/embeddings-transformers#embedding-layers) in the docs, is there a particular part of it you had a question about?

**Reply by svlandeg** [reply]
Update: our coref component is now released in `spacy-experimental`, cf https://github.com/explosion/spaCy/discussions/11585
